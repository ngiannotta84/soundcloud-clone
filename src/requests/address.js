// const address = "https://soundcloud-clone-api.onrender.com";
const address = "http://localhost:4000";

export default address;
