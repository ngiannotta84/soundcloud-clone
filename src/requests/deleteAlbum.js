import { deleteRequest } from "./helpers";

const deleteAlbum = async (id) => {
  const response = await deleteRequest("albums", id);
  return response;
};

export default deleteAlbum;
