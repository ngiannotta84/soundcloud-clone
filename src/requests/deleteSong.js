import { deleteRequest } from "./helpers";

const deleteSong = async (id) => {
  try {
    const response = await deleteRequest("songs", id);
    return response;
  } catch (err) {
    throw new Error(err);
  }
};

export default deleteSong;
