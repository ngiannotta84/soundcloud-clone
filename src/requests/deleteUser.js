import { deleteRequest } from "./helpers";

const deleteUser = async (id, password) => {
  const endpoint = `${id}/${password}`;

  const response = await deleteRequest("users", endpoint);
  return response;
};

export default deleteUser;
