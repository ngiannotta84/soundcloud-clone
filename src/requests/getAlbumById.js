/* eslint-disable no-console */
import axios from "axios";
import address from "./address";

const getAlbumById = async (id) => {
  try {
    const response = await axios.get(`${address}/albums/${id}`);
    return response.data;
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};

export default getAlbumById;
