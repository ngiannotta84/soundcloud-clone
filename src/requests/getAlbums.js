/* eslint-disable no-console */
import axios from "axios";
import address from "./address";

const getAlbums = async ({ name, exact, limit }) => {
  let endpoint = `${address}/albums`;
  const queryArray = [];
  if (name) {
    queryArray.push(`name=${name}`);
    if (exact) {
      queryArray.push("exact=true");
    }
  }
  if (limit) {
    queryArray.push(`limit=${limit}`);
  }
  if (queryArray.length > 0) {
    const query = queryArray.join("&");
    endpoint = `${endpoint}?${query}`;
  }
  console.log(endpoint);
  try {
    const response = await axios.get(endpoint);
    console.log(response);
    return response.data;
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};

export default getAlbums;
