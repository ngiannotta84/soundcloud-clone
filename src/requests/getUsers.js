/* eslint-disable no-console */
import axios from "axios";
import address from "./address";

const getUsers = async (obj) => {
  let endpoint = `${address}/users`;
  const queryArray = [];
  if (obj) {
    if (obj.name) {
      queryArray.push(`name=${obj.name}`);
      if (obj.exact) {
        queryArray.push("exact=true");
      }
    }
    if (obj.limit) {
      queryArray.push(`limit=${obj.limit}`);
    }
  }

  if (queryArray.length > 0) {
    const query = queryArray.join("&");
    endpoint = `${endpoint}?${query}`;
  }

  try {
    const response = await axios.get(endpoint);
    return response.data;
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};

export default getUsers;
