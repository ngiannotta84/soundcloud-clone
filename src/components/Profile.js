import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import PropTypes from "prop-types";
import getUsers from "../requests/getUsers";
import Album from "./Album";

const Profile = ({ handleSetPlaylist }) => {
  const initialState = {
    profile: {
      name: "no user found",
    },
  };
  const { userName } = useParams();
  const [profile, setProfile] = useState(initialState.profile);

  useEffect(() => {
    (async () => {
      try {
        const [response] = await getUsers({ name: userName, exact: true });
        if (response) {
          setProfile(response);
        } else {
          setProfile(initialState.profile);
        }
      } catch (err) {
        setProfile(initialState.profile);
      }
    })();
  }, [userName]);

  return (
    <div>
      <h2 data-testid="profile-name">{profile.name}</h2>
      {profile.Albums &&
        profile.Albums.map((album) => {
          return (
            <Album
              artistName={profile.name}
              albumName={album.name}
              albumArt={album.url}
              songs={album.Songs}
              key={album.id}
              handleSetPlaylist={handleSetPlaylist}
            />
          );
        })}
    </div>
  );
};

Profile.propTypes = {
  handleSetPlaylist: PropTypes.func.isRequired,
};

export default Profile;
