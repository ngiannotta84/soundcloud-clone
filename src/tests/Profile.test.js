import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import Profile from "../components/Profile";
import * as getUsers from "../requests/getUsers";
import fakeUserData from "./testData/fakeUserData";

describe("Profile", () => {
  const validProps = {
    handleSetPlaylist: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(getUsers, "default").mockResolvedValue(fakeUserData);
  });

  test("snapshot", async () => {
    let asFragment;
    await act(() => {
      const view = render(
        <Profile handleSetPlaylist={validProps.handleSetPlaylist} />
      );
      asFragment = view.asFragment;
    });

    expect(asFragment()).toMatchSnapshot();
  });

  describe("truthy response", () => {
    beforeEach(async () => {
      await act(() => {
        render(<Profile handleSetPlaylist={validProps.handleSetPlaylist} />);
      });
    });

    test("renders correctly", async () => {
      await waitFor(() => {
        const images = screen.getAllByRole("img");

        expect(screen.getByTestId("profile-name")).toHaveTextContent(
          fakeUserData[0].name
        );
        expect(images).toHaveLength(fakeUserData[0].Albums.length);
        expect(images[0]).toHaveAttribute(
          "alt",
          `${fakeUserData[0].Albums[0].name} cover art`
        );
      });
    });
  });
});
