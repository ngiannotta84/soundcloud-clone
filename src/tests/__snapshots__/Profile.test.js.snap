// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      data-testid="profile-name"
    >
      name 1
    </h2>
    <div
      class="album"
    >
      <div
        class="album__info-container"
      >
        <img
          alt="album1 cover art"
          class="album__cover-art"
          src="art url 1"
        />
        <div>
          <h2
            class="album__artist-name"
          >
            name 1
          </h2>
          <h3
            class="album__album-name"
          >
            album1
          </h3>
        </div>
      </div>
      <div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song1
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song2
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="album"
    >
      <div
        class="album__info-container"
      >
        <img
          alt="album2 cover art"
          class="album__cover-art"
          src="art url 2"
        />
        <div>
          <h2
            class="album__artist-name"
          >
            name 1
          </h2>
          <h3
            class="album__album-name"
          >
            album2
          </h3>
        </div>
      </div>
      <div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song4
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song3
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
